token/cli/src/main.rs

```diff
@@ -7040,4 +7129,209 @@ mod tests {
+
+        let owner_keypair_file = NamedTempFile::new().unwrap();
+        write_keypair_file(&payer, &owner_keypair_file).unwrap();
+
+        let program_id = &spl_token_2022::id();
+        let config = test_config_with_default_signer(&test_validator, &payer, program_id);
+
+        let multisig = Arc::new(Keypair::new());
+        let multisig_pubkey = multisig.pubkey();
+
+        // add the multisig as a member to itself, make it self-owned
+        let multisig_members = std::iter::once(multisig_pubkey)
+            .chain(multisig_members.iter().cloned())
+            .collect::<Vec<_>>();
+        let multisig_path = NamedTempFile::new().unwrap();
+        write_keypair_file(&multisig, &multisig_path).unwrap();
+        let multisig_paths = std::iter::once(&multisig_path)
+            .chain(multisig_paths.iter())
+            .collect::<Vec<_>>();
+
+        command_create_multisig(&config, multisig, m, multisig_members)
+            .await
+            .unwrap();
+
+        let account = config
+            .rpc_client
+            .get_account(&multisig_pubkey)
+            .await
+            .unwrap();
+        let multisig = Multisig::unpack(&account.data).unwrap();
+        assert_eq!(multisig.m, m);
+        assert_eq!(multisig.n, n);
+
+        let receiver = Keypair::new();
+        let excess_lamports = 4000 * 1_000_000_000;
+
+        config
+            .rpc_client
+            .send_and_confirm_transaction(&Transaction::new_signed_with_payer(
+                &[system_instruction::transfer(
+                    &payer.pubkey(),
+                    &multisig_pubkey,
+                    excess_lamports,
+                )],
+                Some(&payer.pubkey()),
+                &[&payer],
+                config.rpc_client.get_latest_blockhash().await.unwrap(),
+            ))
+            .await
+            .unwrap();
+
+        exec_test_cmd(
+            &config,
+            &[
+                "spl-token",
+                CommandName::WithdrawExcessLamports.into(),
+                &multisig_pubkey.to_string(),
+                &receiver.pubkey().to_string(),
+                "--owner",
+                &multisig_pubkey.to_string(),
+                "--multisig-signer",
+                multisig_paths[0].path().to_str().unwrap(),
+                "--multisig-signer",
+                multisig_paths[1].path().to_str().unwrap(),
+                "--multisig-signer",
+                multisig_paths[2].path().to_str().unwrap(),
+                "--fee-payer",
+                &fee_payer_keypair_file.path().to_str().unwrap(),
+                "--program-id",
+                &program_id.to_string(),
+            ],
+        )
+        .await
+        .unwrap();
+
+        assert_eq!(
+            excess_lamports,
+            config
+                .rpc_client
+                .get_balance(&receiver.pubkey())
+                .await
+                .unwrap()
+        );
```

Comment:
Since you're testing three different cases, you can make three separate tests for them. This seems like a nice break to start the second test